crates/router/src/connector/mifinity/transformers.rs

```diff
@@ -124,109 +278,147 @@ impl<F, T>
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
+        let payload = item.response.payload.first();
+
+        let trace_id = payload.map(|payload| payload.trace_id.clone()).ok_or(
+            errors::ConnectorError::MissingRequiredField {
+                field_name: "trace_id",
+            },
+        )?;
```

Comment:
If `trace_id` is not present, we should not throw error, as we can recover using our payment_id, also trace_id is not optional, this is not the correct way of handling